server/CVCMIServer.cpp

```diff
@@ -305,6 +323,10 @@ bool CVCMIServer::prepareToStartGame()
 		break;
 	}
 	
+	progressTrackingFinished = true;
+	while(progressTrackingFinished)
+		continue;
```

Comment:
WTF?
Is this busy wait?
Is this busy waits that use non-atomic variable?

Please use condition_variable instead. Or explicitly join thread instead of detaching it if this is what is needed here.

Reply:
fixed